### ISSUE TITLE
Cross-user portable way to make PyLance check user Python packages

### ISSUE THREAD

**Issue by metov** [microsoft/pylance-release#3788]
It seems like PyLance is aware of system Python packages (eg. `/usr/lib/python3.11/site-packages/`) but not the user packages (eg. `~/.local/lib/python3.11/site-packages/`). When a package `tqdm` is installed in the user directory (can be requested with `pip install --user`, but pip will fall back to the user dir even without `--user` when it detects that `sudo` is needed for the system dir), PyLance flags `import tqdm` and claims the "package is not found".

The package is obviously there: For example, `pip show tqdm` will succeed and running the program will not generate an `ImportError`, because the `tqdm` is in the Python path.

As a user, I would expect PyLance to check the Python path (for example: https://stackoverflow.com/q/1489599/18571602) to figure out which packages are installed. However, it appears that PyLance does not check the Python path, and instead uses its own guess for where the packages are installed, which turns out to be incorrect.

Perhaps PyLance is already checking the Python path, and something is misconfigured on my end? I would appreciate any pointers here.

Otherwise, I found that the only way to make PyLance aware of my packages is to add the following to `User/settings.json`:
```
"python.analysis.extraPaths": [
    "/home/metov/.local/lib/python3.11/site-packages"
  ],
```

Unfortunately, this does require the explicit path. `~/.local/lib/python3.11/site-packages` does not work. So if you are trying to reuse the same configuration across different machines, this is unusable - it would stop working if any of the machines has a different username or Python version. The `.local/lib` part can also be changed by XDG variables.

Is there a solution to this that I haven't noticed? Or is this intended behavior?

**Comment by erictraut** [comment]
By default, pylance uses the `sys.path` directory list provided by the configured python environment. 

If you open the Output tab in VS Code and select "Python Language Server", you will see Pylance's log output, and it should include the list of directories it is using for import resolutions.

**Comment by metov** [comment]
I tried to come up with steps for reproducing this, and it looks like this issue isn't happening anymore.

1. I deleted `python.analysis.extraPaths` from my config
2. I found a package that is not installed on my system - `permutation` - and confirmed it with `pip show permutation` failing and `import permutation` raising `ModuleNotFoundError`
3. I wrote a simple Python script that imports this, PyLance correctly gives an error when this is open in VSC as per `reportMissingImports`
4. I did `pip install permutation`. Pip warned that it's using the user installation.
5. The test script ran successfully after this
6. PyLance was still giving the error in the window I previously had open - even though the package is now installed
7. After closing VSC and opening it again, the error had gone away

I am guessing that PyLance needs to reindex the packages periodically, and this either happens at program launch or at occasional intervals. Whichever it is, it seems like a reindex is triggered on launch. This is probably why I thought it wasn't working - I wouldn't have tried to restart VSC.

This is with the default interpreter, by the way. `python.defaultInterpreterPath` is set to `python` and `which python` returns `/usr/bin/python` on my system.

Seems like everything is working, so there is no need to fix anything. My bad! :)
